### PR TITLE
DOC: timeseries.rst floating point precision (#15817)

### LINKED ISSUE #15817: to_datetime() ns precision inconsistencies between s and ns
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
pd.options.display.float_format = '{:.20f}'.format
with open('example.csv', 'w') as f:
    f.write('''
full,ts,nano
1490195805.433502912,1490195805,433502912
1490195805.933358907,1490195805,933358907
1490195806.433445930,1490195806,433445930
1490195806.933351039,1490195806,933351039
''')
df = pd.read_csv('example.csv')
df['from two ints'] = pd.to_datetime(df['ts'] * 10**9 + df['nano'], unit='ns')
df['from float to s'] = pd.to_datetime(df['full'], unit='s')
df['from float^9 to ns'] = pd.to_datetime(df['full'] * 10**9, unit='ns')

print('Types:')
print(df.dtypes)
print('Data:')
print(df.iloc[0])
```
Output

```
Types:
full                         float64
ts                             int64
nano                           int64
from two ints         datetime64[ns]
from float to s       datetime64[ns]
from float^9 to ns    datetime64[ns]
dtype: object
Data:
full                 1490195805.43350267410278320312
ts                                        1490195805
nano                                       433502912
from two ints          2017-03-22 15:16:45.433502912
from float to s           2017-03-22 15:16:45.433503
from float^9 to ns     2017-03-22 15:16:45.433502720
Name: 0, dtype: object
```

#### Problem description
I would expect that all of the following methods would give the same result. In the [documentation](http://pandas.pydata.org/pandas-docs/stable/timeseries.html#epoch-timestamps) that the epoch times will be rounded to the nearest nanosecond. I don't understand why conversion from float using seconds `s` unit gives different rounding than nano seconds `ns`. Not to mention strange float parsing coming from csv.

I expect the whole issue to be connected to #7307.

#### Expected Output

```
Types:
full                         float64
ts                             int64
nano                           int64
from two ints         datetime64[ns]
from float to s       datetime64[ns]
from float^9 to ns    datetime64[ns]
dtype: object
Data:
full                            1490195805.433502912
ts                                        1490195805
nano                                       433502912
from two ints          2017-03-22 15:16:45.433502912
from float to s        2017-03-22 15:16:45.433502912
from float^9 to ns     2017-03-22 15:16:45.433502912
Name: 0, dtype: object
```
#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.12.final.0
python-bits: 64
OS: Linux
OS-release: 4.9.10-040910-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: None.None

pandas: 0.19.2
nose: None
pip: 9.0.1
setuptools: 34.3.2
Cython: 0.25.2
numpy: 1.10.4
scipy: 0.19.0
statsmodels: None
xarray: None
IPython: 5.3.0
sphinx: 1.4.8
patsy: None
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: None
tables: 3.3.0
numexpr: 2.6.2
matplotlib: 2.0.0
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.5.3
html5lib: 0.999999999
httplib2: 0.9.1
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.9.5
boto: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by mchwalisz**
- [x] closes #15817

### CODE REVIEW

File: pandas/tseries/tools.py [hunk 1]
```diff
@@ -237,6 +237,10 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
         integer or float number. This will be based off the origin.
         Example, with unit='ms' and origin='unix' (the default), this
         would calculate the number of milliseconds to the unix epoch start.
+
+        Warning: For float arg, precision rounding might happen. To prevent
+        unexpected behavior use a fixed-width exact type.
```
Comment (jreback) [inline pandas/tseries/tools.py]: this is true only for ns (make that clear); i would put this in the Notes section

File: doc/source/timeseries.rst [hunk 2]
```diff
@@ -265,6 +265,22 @@ Typical epoch stored units
    pd.to_datetime([1349720105100, 1349720105200, 1349720105300,
                    1349720105400, 1349720105500 ], unit='ms')
 
+Conversion of float epoch times can lead to inaccurate and unexpected results.
```
Comment (jreback) [inline doc/source/timeseries.rst]: i would make this a note

also units are rounded to the unit precision so the first example is misleading

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tseries/tools.py (hunk 1 above)
Comment (mchwalisz) [inline pandas/tseries/tools.py]: If you mean for `unit='ns'` I don't agree. You can use `unit='s'` and have result saved up to mili seconds with rounding error from float. as in timeseries example

File: doc/source/timeseries.rst (hunk 2 above)
Comment (mchwalisz) [inline doc/source/timeseries.rst]: you mean the line as note or the whole section?

also unit is taken as the base but precision is taken from float. First example is exactly the problem I've had at the beginning.

```
In [36]: pd.to_datetime(1490195805.433502912, unit='s')
Out[36]: Timestamp('2017-03-22 15:16:45.433503')
```
It neither `s` nor `ns` in the example.

Comment (jreback) [inline doc/source/timeseries.rst]: right, that was the example, ok then.

Comment (jreback) [inline doc/source/timeseries.rst]: I mean this whole section  you added to make a note (or could do a warning).

File: doc/source/timeseries.rst [hunk 3]
```diff
@@ -265,6 +265,22 @@ Typical epoch stored units
...
+
+   1490195805.433502912
+
+   1490195805433502912
+
+   pd.to_datetime(1490195805.433502912, unit='s')
+
+   pd.to_datetime(1490195805433502912, unit='ns')
+
 These *work*, but the results may be unexpected.
```
Comment (jreback) [inline doc/source/timeseries.rst]: you may have to adjust this line after

File: pandas/tseries/tools.py (hunk 1 above)
Comment (jreback) [inline pandas/tseries/tools.py]: right. ok in any even still put it in the Notes section.

**Comment by mchwalisz** [comment]
I've moved the warning in `to_datetime` to examples and rearranged sections in `timeseries.rst` to make it more logical (IMO).

**Comment by jreback** [comment]
lgtm. @jorisvandenbossche

File: doc/source/timeseries.rst [hunk 4]
```diff
@@ -265,17 +265,34 @@ Typical epoch stored units
...
-These *work*, but the results may be unexpected.
+.. note::
 
-.. ipython:: python
+   Epoch times will be rounded to the nearest nanosecond.
+
+.. warning::
+
+   Conversion of float epoch times can lead to inaccurate and unexpected results.
+   Python floats are unbounded precision, with about 15 significant digits (depends on
```
Comment (jorisvandenbossche) [inline doc/source/timeseries.rst]: What do you mean with "unbounded precision"? that seems to contradict a bit with "15 digits"

File: doc/source/timeseries.rst [hunk 5]
```diff
@@ -265,17 +265,34 @@ Typical epoch stored units
...
+   operating system). Rounding in conversion from to high precision ``Timestamp`` is
```
Comment (jorisvandenbossche) [inline doc/source/timeseries.rst]: There seems to be something wrong in the "in conversion from to high precision ``Timestamp``" part (it doesn't read fluent to me)

**Review by jorisvandenbossche — COMMENTED** [review]
Some small comments about the wording (to make sure everything is clear to the reader!)

File: doc/source/timeseries.rst (hunk 5 above)
Comment (mchwalisz) [inline doc/source/timeseries.rst]: flaot is missing. will correct it in a sec.

Comment (jreback) [inline doc/source/timeseries.rst]: pls emphasize that this is only on input of *very* high precision float data.

File: doc/source/timeseries.rst [hunk 6]
```diff
@@ -265,17 +265,34 @@ Typical epoch stored units
...
-   pd.to_datetime([1, 3.14], unit='s')
+      1490195805.433502912
+      pd.to_datetime([1490195805.433, 1490195805.433502912], unit='s')
+      1490195805433502912
+      pd.to_datetime(1490195805433502912, unit='ns')
 
 .. note::
 
-   Epoch times will be rounded to the nearest nanosecond.
+   These *work*, but the results may be unexpected.
```
Comment (jreback) [inline doc/source/timeseries.rst]: this part is now hanging, so add something like. numeric input is in referece to the ``epoch`` time (actually may want to move this mini-example (just this last part here) to the 'origin' section.)

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**